recipes/gperftools/all/conanfile.py

```diff
@@ -0,0 +1,72 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+
+class GperftoolsConan(ConanFile):
+    name = "gperftools"
+    license = "BSD-3-Clause License"
+    homepage = "https://github.com/gperftools/gperftools"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "gperftools: originally Google Performance Tools"
+    topics = ("malloc", "profiler")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"fPIC": False, "shared": True}
+    autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self.autotools is None:
+            autotools = AutoToolsBuildEnvironment(self)
+            with tools.chdir(self._source_subfolder):
+                self.run("autoreconf -fiv")
+            default_args = []
+            if self.options.shared:
+                default_args.append("--disable-static")
+            else:
+                default_args.append("--disable-shared")
+            if self.options.fPIC:
+                default_args.append("--with-pic=yes")
+            autotools.configure(
+                configure_dir=self._source_subfolder,
+                args=default_args
+            )
+            self.autotools = autotools
+        return self.autotools
```

Comment:
fPIC is handled automatically by the conan client.
But to be sure that static libraries are compiled with fpic when compiled as a shared library, set it explicitly.
Also, only run autoreconf only in build, before `_configure_autotools`.
```suggestion
    @functools.lru_cache(1)
    def _configure_autotools(self):
        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
        yes_no = lambda v: "yes" if v else "no"
        args = [
            f"--enable-shared={yes_no(self.options.shared)}",
            f"--enable-static={yes_no(not self.options.shared)}",
        ]
        autotools.configure(args=args, configure_dir=self._source_subfolder)
        autotools.fpic = self.options.get_safe("fPIC", True)
        return autotools
```